service-workers/service-worker/multi-globals/url-parsing.https.html

```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE html>
+<title>register()/getRegistration() URL parsing, with multiple globals in play</title>
+<link rel="help" href="https://w3c.github.io/ServiceWorker/spec/service_worker/index.html#service-worker-container-register-method">
+<link rel="help" href="https://w3c.github.io/ServiceWorker/spec/service_worker/index.html#service-worker-container-getregistration-method">
+<link rel="author" title="Domenic Denicola" href="mailto:d@domenic.me">
+<script src="/resources/testharness.js"></script>
+<script src="../resources/testharness-helpers.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="../resources/test-helpers.sub.js"></script>
+
+<!-- This is the entry global -->
+
+<iframe src="incumbent/incumbent.https.html"></iframe>
+
+<script>
+'use strict';
+
+const loadPromise = new Promise(resolve => {
+    window.addEventListener('load', () => resolve());
+});
+
+promise_test(t => {
+    let registration;
+
+    return loadPromise.then(() => {
+        return frames[0].testRegister();
+    }).then(r => {
+        registration = r;
+        return fetch('test.txt');
```

Comment:
This fetch is not intercepted as the document hasn't gotten its controller yet. We should make sure the registration has advanced to "activated" state before this fetch. For that, add a promise returning utility function [wait_for_state](https://github.com/w3c/web-platform-tests/blob/master/service-workers/service-worker/resources/test-helpers.sub.js#L78) before this line. It would look like:
```js
const newestWorker = r.installing || r.waiting || r.active;
return wait_for_state(newestWorker, 'activated').then(_ => {
  // Fetch here
});
```

Also, the following code is needed in the service worker script to promote the worker to active worker and to make all of its clients controlled right away (without reloading them:)
```js
this.addEventListener('install', event => {
    this.skipWaiting();
});

this.addEventListener('activate', event => {
    clients.claim();
});
```

And I think the fetch should `fetch('relevant/test.txt')` instead of `test.txt` as we expect the relevant scoped worker intercept this request.